src/web/app/components/question-types/question-statistics/question-statistics-calculation/rank-recipients-question-statistics-calculation.ts

```diff
@@ -57,13 +66,26 @@ export class RankRecipientsQuestionStatisticsCalculation
       if (!this.emailToName[identifier]) {
         this.emailToName[identifier] = response.recipient;
       }
+
+      optionsPerTeam[response.recipientTeam] = optionsPerTeam[response.recipientTeam] || [];
```

Comment:
the naming here is a little confusing - we are referring to `recipient` but the variable names all somehow say `option`. 

I think we might be able to streamline this entire statistic calculation with a combination of `map`, `filter` and `reduce` (in a functional manner without for loops).  However, this I leave it up to you: as long as it's readable I am okay.